### DISCUSSION TITLE
InitialValue Of a Field

### DISCUSSION THREAD

**Discussion by dashne** [Washi1337/AsmResolver#596] [Q&A] [answered]
Hi everyone, How to Get Initial Value of a field, if it has it? please provide sample code for demonstration.

Thanks

**Comment by Washi1337** [comment, marked as answer]
The initial value for fields are stored in a FieldRva row, which can be obtained by the `FieldRva` property: 

```csharp
FieldDefinition field = ...;

if (field.FieldRva is ISegment segment)
{
   // Field has FieldRva data (initial value).
   byte[] rawData = segment.WriteIntoArray();
}
```

**Reply by dashne** [reply]
Thank you so much
